### DISCUSSION TITLE
Less bot more org

### DISCUSSION THREAD

**Discussion by fcollonval** [jupyterlab-contrib/jupyterlab-contrib.github.io#46] [General]
Related to https://github.com/jupyterlab-contrib/jupyterlab-contrib.github.io/issues/42, I created a PyPI organization - https://pypi.org/org/jupyterlab-contrib/ to ease right management on PyPI and drop the need for the bot account.

The end goal is to get one place per platform to manage rights on GitHub, PyPI and NPM (conda-forge being its own beast) and remove the use of a bot account (it is for now still needed for NPM and for GitHub when using the releaser).

Therefore I would like to align the roles by promoting @jtpio @krassowski @martinRenou and myself  to be owners on the three platforms.

> Other members will be invited to become members of the PyPI organization.

Here are my questions:

- Do Jeremy, Mike and Martin agree to become owner on the three platforms?
- Does someone else is interested by being granted more rights (we should keep the number of owners between 3 and 5 to avoid bottlenecks while limiting the number for security reasons)?

> Jeremy, Mike and Martin, I'm gonna need your help to add some project to the PyPI organization as I don't have enough rights on: 
- [x] [jupyterlab-pytutor](https://pypi.org/project/jupyterlab-pytutor/), 
- [x] [jupyterlab-spreadsheet-editor](https://pypi.org/project/jupyterlab-spreadsheet-editor/), 
- [x] [jupyterlab-spellchecker](https://pypi.org/project/jupyterlab-spellchecker/), 
-  ~~[ipylgbst](https://pypi.org/project/ipylgbst/)~~ (migrated to another organization),
- [x] [jupyterlab-fileopen](https://pypi.org/project/jupyterlab-fileopen/)

---

cc @jupyterlab-contrib/members @bollwyvl @ianhi @marimeireles @mlhenderson @ocordes @wolfv @zacharyrs

**Comment by jtpio** [comment]
Thanks @fcollonval.

> Do Jeremy, Mike and Martin agree to become owner on the three platforms?

Yes :+1: 

Happy to do the transfer of `jupyterlab-pytutor` after getting the rights.

For `ipylgbst` it will likely be moved to another org: https://github.com/jupyter-widgets-contrib/jupyter-widgets-contrib.github.io/issues/1

**Comment by fcollonval** [comment]
Thanks for the feedback and for the good news about jupyter-widget-contrib :tada: 

GitHub and NPM rights have been updated for the four owners and owners should have received invitation to join the PyPI organization.

I would like also to ask all owners to activate 2FA on the three platforms ([GitHub](https://docs.github.com/en/authentication/securing-your-account-with-two-factor-authentication-2fa/configuring-two-factor-authentication), [NPM](https://docs.npmjs.com/configuring-two-factor-authentication) and [PyPI](https://pypi.org/help/#twofa)).

**Reply by krassowski** [reply]
Can we enforce 2FA?

**Reply by fcollonval** [reply]
Thanks for the question. I'll open a separate discussion for this one.

**Comment by jtpio** [comment]
`jupyterlab-pytutor` has been transferred to the `jupyterlab-contrib` org on PyPI: 

![image](https://github.com/jupyterlab-contrib/jupyterlab-contrib.github.io/assets/591645/c85c8938-db2b-44bb-9afc-c7454c08a614)

**Comment by fcollonval** [comment]
Thanks to all owners for moving the remaining packages to the new PyPI org. I'll close this one and open a separate thread about 2FA enforcement.
